### DISCUSSION TITLE
Comparison Signal Recived in elsa 2 to elsa 3

### DISCUSSION THREAD

**Discussion by nurhidayat-agung** [elsa-workflows/elsa-core#5613] [Q&A]
hello elsa ?

is there any activities that have same fucntion to signal recived like in elsa 2 in elsa 3 ?

**Comment by glime-ai** [comment]
In Elsa 3, the activities that were known as "Send Signal" and "Signal Received" in Elsa 2 have been replaced with different concepts and activities. The signaling mechanism in Elsa 3 is not the same as in Elsa 2. Instead of "Send Signal" and "Signal Received," you might use the "Publish Event" and "Event" activities for similar purposes in Elsa 3.

The "Publish Event" activity allows you to publish an event with a name and correlation ID, while the "Event" activity acts as an event handler that can continue the workflow if an event with the specified name and correlation ID is raised.

If you are looking to implement custom logic before suspending the workflow, you may need to create custom activities. For more information on the available activities in Elsa 3 and how to use them, you can refer to the official Elsa documentation, specifically the sections on primitive activities, flow activities, and branching activities. If you need to create custom activities, you can follow the guidelines provided in the documentation and examples from the community.

For those migrating from Elsa 2 to Elsa 3 and needing specific guidance on how to replace the "Send Signal" and "Signal Received" activities, it would be best to consult the Elsa 3 documentation or reach out to the Elsa community for advice on how to achieve similar functionality with the new version.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
